### PR TITLE
fix: fb image payload comment

### PR DESCRIPTION
**Pull request by zhouhanseng**
close https://github.com/DimensionDev/Maskbook/issues/1957

### CODE REVIEW

File: packages/maskbook/src/social-network-provider/facebook.com/UI/collectPosts.tsx [hunk 1]
```diff
@@ -86,14 +88,21 @@ export function collectPostsFacebook(this: SocialNetworkUI) {
...
+                    })
+
+                    if (post.indexOf('🎼') === 0 || /https:\/\/.+\..+\/(\?PostData_v\d=)?%20(.+)%40/.test(post)) {
+                        info.iv.value = deconstructPayload(post, payloadDecoder).unwrap().iv
+                    }
                 }
                 // parse post content
                 info.postMessage.value = makeTypedMessageCompound(nextTypedMessage)
             }
             collectPostInfo()
```
Comment (septs) [inline packages/maskbook/src/social-network-provider/facebook.com/UI/collectPosts.tsx:102]: await?

Comment (zhouhanseng) [reply packages/maskbook/src/social-network-provider/facebook.com/UI/collectPosts.tsx:102]: ` new MutationObserverWatcher(posts).useForeach(async (node, key, metadata) => { //...` is not allowed, so here can't use `await collectPostInfo()`, we can do more investigation later. Currently fix the bug first.

Comment (zhouhanseng) [reply packages/maskbook/src/social-network-provider/facebook.com/UI/collectPosts.tsx:102]: https://git.io/JkFKT How about make `foreach()` async

Comment (Jack-Works) [reply packages/maskbook/src/social-network-provider/facebook.com/UI/collectPosts.tsx:102]: No need to await, add `.catch(console.error)` is enough for me

File: packages/maskbook/src/social-network/PostInfo.ts [hunk 2]
```diff
@@ -52,6 +52,7 @@ export abstract class PostInfo {
     readonly decryptedPostContent = new ValueRef<TypedMessage | null>(null)
     /** @deprecated It should appear in the transformedPostContent */
     readonly decryptedPostContentRaw = new ValueRef('')
+    readonly iv = new ValueRef('')
```
Comment (Jack-Works) [inline packages/maskbook/src/social-network/PostInfo.ts:55]: It should be `ValueRef<null | string>`, don't use an empty string as empty state

File: packages/maskbook/src/plugins/Wallet/UI/WalletStatusDialog.tsx [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+        undefined,
+        undefined,
+        t('copy_success_of_wallet_addr'),
+    )
+    //#endregion
+
+    //#region remote controlled dialog logic
+    const [open, setWalletStatusDialogOpen] = useRemoteControlledDialog(
+        WalletMessageCenter,
+        'walletStatusDialogUpdated',
```
Comment (Jack-Works) [inline packages/maskbook/src/plugins/Wallet/UI/WalletStatusDialog.tsx:85]: Are you sure your code is up to date? IMO this is the old pattern.

File: package.json [hunk 4]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "maskbook",
-  "version": "1.19.4",
+  "version": "1.19.5",
```
Comment (guanbinrui) [inline package.json:3]: Revert this.

**Comment by guanbinrui** [comment]
@zhouhanseng Plz rebasing the last commit to the released branch.

**Comment by septs** [comment]
Plz rebase master (resolve conflict)

File: packages/maskbook/src/social-network/PostInfo.ts (hunk 2 above)
Comment (zhouhanseng) [reply packages/maskbook/src/social-network/PostInfo.ts]: done

File: packages/maskbook/src/social-network-provider/facebook.com/UI/collectPosts.tsx (hunk 1 above)
Comment (zhouhanseng) [reply packages/maskbook/src/social-network-provider/facebook.com/UI/collectPosts.tsx]: done

**Comment by zhouhanseng** [comment]
@guanbinrui done

**Comment by guanbinrui** [comment]
> @guanbinrui done

It looks like you pushed to `hotfix/1.19.6`. I re-picked you commits to `hotfix/1.19.6` and force pushed to `hancheng/crypto-comment-img-payload`. Plz have a check.

File: packages/maskbook/src/social-network-provider/facebook.com/UI/collectPosts.tsx [hunk 5]
```diff
@@ -86,14 +88,21 @@ export function collectPostsFacebook(this: SocialNetworkUI) {
                 for (const url of images) {
                     info.postMetadataImages.add(url)
                     nextTypedMessage.push(makeTypedMessageImage(url))
+                    const post = await Services.Steganography.decodeImageUrl(url, {
```
Comment (guanbinrui) [inline packages/maskbook/src/social-network-provider/facebook.com/UI/collectPosts.tsx:91]: This is not right for me.

Comment (zhouhanseng) [reply packages/maskbook/src/social-network-provider/facebook.com/UI/collectPosts.tsx:91]: Why?

Comment (guanbinrui) [reply packages/maskbook/src/social-network-provider/facebook.com/UI/collectPosts.tsx:91]: The collector should do its best on collecting info from the DOM nodes. Besides that, the decryption progress was handled by the  `<DecryptedPost />` component. And you should not do it twice here. The raw `decodeImageUrl` service did not cache the result for you which is already supported by the `<DecryptedPost />` component.

**Comment by zhouhanseng** [comment]
I prefer to open a new pr based on master branch, since I think it is not just a hot fix.

File: packages/maskbook/src/social-network-provider/facebook.com/UI/collectPosts.tsx (hunk 5 above)
Comment (zhouhanseng) [reply packages/maskbook/src/social-network-provider/facebook.com/UI/collectPosts.tsx:91]: Yeah, set the iv of post during decrypt image.

**Comment by zhouhanseng** [comment]
move to https://github.com/DimensionDev/Maskbook/pull/1971

**Pull request closed without merging**